### PR TITLE
feat: provide onlyShowDefault option and expose getSystemSelectionQuestions

### PR DESCRIPTION
**Pull request by mmilko01**
#2719

- Provides a `onlyShowDefault` option for system selection prompt.
- Exposes getSystemSelectionQuestions

### CODE REVIEW

File: packages/odata-service-inquirer/src/index.ts [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 import { type Logger } from '@sap-ux/logger';
 import { OdataVersion } from '@sap-ux/odata-service-writer';
 import { type ToolsSuiteTelemetryClient } from '@sap-ux/telemetry';
 import autocomplete from 'inquirer-autocomplete-prompt';
 import { ERROR_TYPE, ErrorHandler, setTelemetryClient } from '@sap-ux/inquirer-common';
 import { initI18nOdataServiceInquirer } from './i18n';
 import { getQuestions } from './prompts';
-import { SystemSelectionAnswerType } from './prompts/datasources/sap-system/system-selection';
+import {
+    type SystemSelectionAnswers,
```
Comment (IainSAP) [inline packages/odata-service-inquirer/src/index.ts:11]: Unused import?

File: packages/odata-service-inquirer/src/types.ts [hunk 2]
```diff
@@ -262,6 +262,11 @@ export type SystemSelectionPromptOptions = {
      *
      */
     defaultChoice?: string;
+    /**
+     * Only show the default choice in the system selection prompt, this is used to skip the system selection prompt if the default choice is already known.
+     * If the default choice is not known, the prompt will be shown as normal.
```
Comment (IainSAP) [inline packages/odata-service-inquirer/src/types.ts:267]:
```suggestion
     * If the `defaultChoice` value is not found in the systems choices, or the `defaultChoice` option is not specified ,the prompt will be shown as normal.
```

File: packages/odata-service-inquirer/src/index.ts (hunk 1 above)
Comment (mmilko01) [reply packages/odata-service-inquirer/src/index.ts:11]: If you are referring to the import the appears deleted in the diff, it was not deleted. You could see it in line 12.

File: packages/odata-service-inquirer/src/types.ts (hunk 2 above)
Comment (mmilko01) [reply packages/odata-service-inquirer/src/types.ts]: Fixed in: af894387c3637e27cb722f67d25f72fcbc884114

**Review by IainSAP — COMMENTED** [review]
Couple of minor comments thx @mmilko01

File: packages/odata-service-inquirer/src/prompts/datasources/sap-system/service-selection/questions.ts [hunk 3]
```diff
@@ -58,8 +58,10 @@ export function getSystemServiceQuestion(
     const serviceSelectionPromptName = `${promptNamespace}:${promptNames.serviceSelection}`;
 
     let systemServiceQuestion = {
-        when: (): boolean =>
-            connectValidator.validity.authenticated || connectValidator.validity.authRequired === false,
+        when: promptOptions?.hide
```
[the remark this answers is not in the document]
Comment (mmilko01) [reply packages/odata-service-inquirer/src/prompts/datasources/sap-system/service-selection/questions.ts]: I moved the condition system selection as required. The only change to your proposal I did was to extend the condition to check if `promptOptions?.serviceSelection?.hide` is undefined also.

Fixed in: 8cf05c6d9365d40796c00c5fff002949e3309505

File: packages/odata-service-inquirer/test/unit/index-api.test.ts [hunk 4]
```diff
@@ -10,6 +11,11 @@ jest.mock('../../src/prompts', () => ({
     ...jest.requireActual('../../src/prompts')
 }));
 
+jest.mock('../../src/prompts/datasources/sap-system/system-selection', () => ({
+    __esModule: true, // Workaround to for spyOn TypeError: Jest cannot redefine property
```
[the remark this answers is not in the document]
Comment (mmilko01) [reply packages/odata-service-inquirer/test/unit/index-api.test.ts]: Copy/Paste problem, it was like this in the other places in the same file. I fixed all of them.

Fixed in: 8cf05c6d9365d40796c00c5fff002949e3309505

File: packages/odata-service-inquirer/src/prompts/datasources/sap-system/system-selection/questions.ts [hunk 5]
```diff
@@ -113,17 +113,18 @@ export async function getSystemSelectionQuestions(
...
-        ...withCondition(
-            getSystemServiceQuestion(
-                connectValidator,
-                systemSelectionPromptNamespace,
-                promptOptions?.serviceSelection
-            ) as Question[],
-            (answers: Answers) => (answers as SystemSelectionAnswers).systemSelection?.type !== 'newSystemChoice'
-        )
-    );
+    if (typeof promptOptions?.serviceSelection?.hide === 'undefined' || !promptOptions?.serviceSelection?.hide) {
```
Comment (IainSAP) [inline packages/odata-service-inquirer/src/prompts/datasources/sap-system/system-selection/questions.ts:116]: I dont think this additional condition is necessary as:
```
!undefined === true => true
!prompt.undefined === true => true
!prompt.serviceSelection.undefined === true => true
```

Comment (mmilko01) [reply packages/odata-service-inquirer/src/prompts/datasources/sap-system/system-selection/questions.ts]: I'm not even sure how I missed that when I was writing it :). Thanks!

Fixed in: 7635730cc925e9187cf50122427120a32dddda80

File: .changeset/plenty-owls-move.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+"@sap-ux/odata-service-inquirer": patch
```
Comment (IainSAP) [inline .changeset/plenty-owls-move.md:2]: Could we bump minor as there are new APIs and options please?

**Review by IainSAP — APPROVED** [review]
Thanks @mmilko01 LGTM

- New option added, backward compatible updates
- Tests added to cover
- Cset added correctly (minor version to indicate new API and options)

File: packages/odata-service-inquirer/src/types.ts [hunk 7]
```diff
@@ -262,6 +262,12 @@ export type SystemSelectionPromptOptions = {
      *
      */
     defaultChoice?: string;
+    /**
+     * Only show the default choice in the system selection prompt, this is used to skip the system selection prompt if the default choice is already known.
+     * If the `defaultChoice` value is not found in the systems choices, or the `defaultChoice` option is not specified,
+     * this option will not be applied and the full list of choices will be presented o the user.
```
Comment (testojs) [inline packages/odata-service-inquirer/src/types.ts:268]:
```suggestion
     * this option will not be applied and the full list of choices will be presented to the user.
```

Comment (mmilko01) [reply packages/odata-service-inquirer/src/types.ts]: Fixed in: beb38a9e9680bd34b088cf8ba5db0628e3e70c0d

**Review by IainSAP — APPROVED** [review]
Re-approving

**Review by testojs — APPROVED** [review]
* code changes look good
* change set OK
* good test coverage
* not tested manually

**Pull request merged**